discord/ext/commands/context.py

```diff
@@ -456,6 +456,84 @@ def me(self) -> Union[Member, ClientUser]:
         # bot.user will never be None at this point.
         return self.guild.me if self.guild is not None else self.bot.user  # type: ignore
 
+    @discord.utils.cached_property
+    def permissions(self) -> Permissions:
+        """:class:`.Permissions`: Returns the resolved permissions for the invoking user in this channel.
+        Shorthand for :meth:`.abc.GuildChannel.permissions_for` or :attr:`.Interaction.permissions`.
+
+        .. versionadded:: 2.0
+        """
+        if self.channel.type is ChannelType.private:
+            return Permissions._dm_permissions()
+        if not self.interaction:
+            # channel and author will always match relevant types here
+            return self.channel.permissions_for(self.author)  # type: ignore
+        base = self.interaction.permissions
+        if self.channel.type in (ChannelType.voice, ChannelType.stage_voice):
+            if not base.connect:
+                # voice channels cannot be edited by people who can't connect to them
+                # It also implicitly denies all other voice perms
+                denied = Permissions.voice()
+                denied.update(manage_channels=True, manage_roles=True)
+                base.value &= ~denied.value
+        else:
+            # text channels do not have voice related permissions
+            denied = Permissions.voice()
+            base.value &= ~denied.value
+        return base
+
+    @discord.utils.cached_property
+    def bot_permissions(self) -> Permissions:
+        """:class:`.Permissions`: Returns the resolved permissions for the bot in this channel.
+
+        For interaction-based commands, this will reflect the effective permissions
+        for :class:`Context` calls, which may differ from calls through
+        other :class:`.abc.Messageable` endpoints, like :attr:`channel`.
+
+        Notably, sending messages, embedding links, and attaching files are always
+        permitted, while reading messages might not be. Mass mentioning and using
+        external emojis and stickers also inherits their permissions from the
+        default guild role.
```

Comment:
This will actually no longer be true in a month.

Reply:
Yeah I think forward compatibility is better here

Reply:
Since the planned change to ATTACH_FILES and EMBED_LINKS was reverted, I've left those in, but otherwise I've now updated this property with the planned 28 July change.